### PR TITLE
Remove name as required prop and add theme to search box

### PR DESCRIPTION
**Pull request by reneedraper**
fix(TextInput): Removed "name" as required prop for textInput components. Allowed searchBox icon, placeholder, border, and value color to be changed by theme. Added darkTheme to searchBox stories.

### CODE REVIEW

File: src/components/SearchBox/SearchBox.js [hunk 1]
```diff
@@ -10,25 +10,26 @@ import _ from 'lodash';
 const SearchBarContainer = styled.div`
     height: 36px;
     border: 1px solid;
-    border-color: ${(props) => { return props.theme.foreground || colors.white60; }};
-    background: ${colors.white10};
+    border-color: ${(props) => { return props.theme.borderColor || colors.white60; }};
```
[the remark this answers is not in the document]
Comment (reneedraper) [reply src/components/SearchBox/SearchBox.js]: Good call

[the remark this answers is not in the document]
Comment (reneedraper) [reply src/components/SearchBox/SearchBox.js]: Should I revert the themes too?

**Comment by aar0nwils0n** [comment]
Could you post a few screenshots of the changes?

**Comment by reneedraper** [comment]
> Could you post a few screenshots of the changes?

image

or maybe 

image

**Comment by aar0nwils0n** [comment]
I believe the dark theme is supposed to look more like this
image
****

**Comment by reneedraper** [comment]
> I believe the dark theme is supposed to look more like this
> image

Right, the only difference is the background.
image

**Comment by reneedraper** [comment]
> I believe the dark theme is supposed to look more like this
> image

The only difference is the background. The border is white40, so you can't really distinguish it from the white background.  I just posted the pink one because I thought you wanted screenshots of how it could change, not screenshots of the actual dark theme. My bad.

**Comment by reneedraper** [comment]
> > I believe the dark theme is supposed to look more like this
> > image
> 
> The only difference is the background. The border is white40, so you can't really distinguish it from the white background. I just posted the pink one because I thought you wanted screenshots of how it could change, not screenshots of the actual dark theme. My bad.

image

**Comment by thundercat1** [comment]
Just curious, what motivated the change to no longer require "name" prop? Like, what harm did it cause? I'm not 100% sure but I think leaving it off could make it tougher for QA to write automated tests against the input field.

**Comment by reneedraper** [comment]
> Just curious, what motivated the change to no longer require "name" prop? Like, what harm did it cause? I'm not 100% sure but I think leaving it off could make it tougher for QA to write automated tests against the input field.

I changed default prop.name to from 'name' to undefined to avoid instances where the label is 'name' and you have to define a name for every text input to avoid jumping back to 'name'. It's no longer a required prop because of that and I got sick of seeing **Warning: Failed prop type: The prop `name` is marked as required in `TextInputBox`, but its value is `undefined`.**

**Pull request merged**